### PR TITLE
Issue 546

### PR DESCRIPTION
**Pull request by mickaelistria**

### CODE REVIEW

File: org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/registry/ProjectFacadeAdapterFactory.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ *******************************************************************************/
+package org.eclipse.m2e.core.internal.project.registry;
+
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.IAdapterFactory;
+
+import org.eclipse.m2e.core.MavenPlugin;
+import org.eclipse.m2e.core.project.IMavenProjectFacade;
+
+public class ProjectFacadeAdapterFactory implements IAdapterFactory {
```
[the remark this answers is not in the document]
Comment (laeubi) [reply org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/registry/ProjectFacadeAdapterFactory.java:23]: This is just run in the IDE (so check in the xml) but you can always write it by hand if you like, it just becomes quite cumbersome to update on changes. Beside this the xml generation works quite good, I can't remember any serve issues in the past years.

File: m2e-maven-runtime/org.eclipse.m2e.maven.runtime/maven.deps [hunk 2]
```diff
@@ -0,0 +1,42 @@
+
```
Comment (laeubi) [inline m2e-maven-runtime/org.eclipse.m2e.maven.runtime/maven.deps:1]: Is this intentional?

Comment (mickaelistria) [reply m2e-maven-runtime/org.eclipse.m2e.maven.runtime/maven.deps:1]: Nope.

Comment (mickaelistria) [reply m2e-maven-runtime/org.eclipse.m2e.maven.runtime/maven.deps]: removed

**Review by laeubi — APPROVED** [review]
Looks good.

**Pull request merged**